sphinx/tutorial/cylc/furthertopics/clock-triggered-tasks.rst

```diff
@@ -0,0 +1,179 @@
+.. include:: ../../../hyperlinks.rst
+   :start-line: 1
+
+Clock Triggered Tasks
+=====================
+
+This part of the Rose user guide walks you through using clock triggered tasks.
+
+This allows you to trigger tasks based on the actual time.
+
+
+Purpose
+-------
+
+.. TODO - Insert animation / diagram.
+
+The typical use of clock triggered tasks is to perform real-time data
+retrieval.
+
+In general, most tasks in a suite do not care about the wall clock time,
+simply running once their prerequisites are met. Clock triggered tasks,
+however, wait for a particular wall clock time to be reached in addition
+to their other prerequisites.
+
+
+Clock Triggering
+----------------
+
+When clock triggering tasks we can use different
+:ref:`offsets <tutorial-iso8601-durations>` to the cycle time as follows:
+
+.. code-block:: cylc
+
+   clock-trigger = taskname(CYCLE_OFFSET)
+
+.. note::
+
+   Regardless of the offset used, the task still belongs to the cycle from
+   which the offset has been applied.
+
+
+Example
+-------
+
+Our example suite will simulate a clock chiming on the hour.
+
+Within your ``~/cylc-run`` directroy create a new directory called
+``clock-trigger``::
+
+   mkdir ~/cylc-run/clock-trigger
+   cd ~/cylc-run/clock-trigger
+
+Paste the following code into a ``suite.rc`` file:
+
+.. code-block:: cylc
+
+   [cylc]
+       UTC mode = True # Ignore DST
+
+   [scheduling]
+       initial cycle point = TODO
+       final cycle point = +P1D # Run for one day
+       [[dependencies]]
+           [[[PT1H]]]
+               graph = bell
```

Comment:
Following on from my comment above, it would likely be helpful to incorporate one or two very simple (normal i.e. non-clock) triggers for 'bell' to the practical in order to emphasise that the clock triggers are usually not the only trigger but combine with standard graph dependencies.

Reply:
My apologies, I have just noticed that there is a very simple, but non-singular, graph in the 'further clock triggering' section (bell => quarter_past => half_past => quarter_to) so the above comment is in a sense addressed, however it still seems trivial since the clock trigger would cause these tasks to run in order anyway, so that the dependency chain itself is already encapsulated and therefore inconsequential - unless I am missing something or misunderstanding here? One or two dependencies that do have an influence as genuine prereqs would more fully illustrate the clock-triggering concept, in my opinion.

Reply:
As discussed offline, this would require coming up with one or more viable dependencies given the 'chiming clock' context. Some (not great but at least I hope vaguely sensible) ideas I can come up with are: a clock that needs winding up or somehow reloading or similar in order for the chime function to work; a bell-ringing scenario whereby the main task is a primary quiet chime that the bell-ringers use as a collective cue to start pulling their ropes to ring the church bells and the dependency is that all the bell-ringers must be in place by their ropes (it is slightly contrived, yes); or maybe a change from a simple chiming scenario to one whereby someone has to check-up on something every hour etc, as prompted by the chiming - and something else is needed to do those checks. I will keep thinking about this - naturally it is much easier to provide criticism than to respond to it!